src/ripple/basics/RangeSet.h

```diff
@@ -20,72 +20,44 @@
 #ifndef RIPPLE_BASICS_RANGESET_H_INCLUDED
 #define RIPPLE_BASICS_RANGESET_H_INCLUDED
 
+#include <boost/icl/closed_interval.hpp>
+#include <boost/icl/interval_set.hpp>
 #include <cstdint>
-#include <map>
-#include <string>
-
-namespace ripple {
 
+namespace ripple
+{
 /** A sparse set of integers. */
-// VFALCO TODO Replace with juce::SparseSet
 class RangeSet
 {
 public:
-    static const std::uint32_t absent = static_cast <std::uint32_t> (-1);
+    static const std::uint32_t absent = static_cast<std::uint32_t>(-1);
 
-public:
-    RangeSet () { }
+    RangeSet() = default;
 
-    bool hasValue (std::uint32_t) const;
+    bool hasValue(std::uint32_t) const;
 
-    std::uint32_t getFirst () const;
-    std::uint32_t getNext (std::uint32_t) const;
-    std::uint32_t getLast () const;
-    std::uint32_t getPrev (std::uint32_t) const;
+    // First number in the set
+    std::uint32_t getFirst() const;
 
-    // largest number not in the set that is less than the given number
-    std::uint32_t prevMissing (std::uint32_t) const;
+    // Largest number not in the set that is less than the given number
+    std::uint32_t prevMissing(std::uint32_t) const;
 
     // Add an item to the set
-    void setValue (std::uint32_t);
+    void setValue(std::uint32_t);
```

Comment:
I know you went out of your way to not rename things, but since you're here, I think that some renaming makes sense: I'd suggest that `hasValue` should be `contains`, `setValue` and `setRange` should really be `add` and `clearValue` should be `remove`.